connaisseur/validators/cosign/cosign_validator.py

```diff
@@ -22,9 +22,10 @@ class CosignValidator(ValidatorInterface):
     name: str
     trust_roots: list
 
-    def __init__(self, name: str, trust_roots: list, **kwargs):
+    def __init__(self, name: str, trust_roots: list, auth: dict = {}, **kwargs):
         super().__init__(name, **kwargs)
         self.trust_roots = trust_roots
+        self.k8s_keychain = auth.get("k8s_keychain", False)
```

Comment:
function defaults should not be [empty mutables in Python](https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python).
```suggestion
    def __init__(self, name: str, trust_roots: list, auth: dict = None, **kwargs):
        super().__init__(name, **kwargs)
        self.trust_roots = trust_roots
        self.k8s_keychain = False if auth is None else auth.get("k8s_keychain", False)
```

Reply:
should be tested when instantiating the class: https://github.com/sse-secure-systems/connaisseur/pull/551/files#diff-60bd9ac24e4f4d47b21f0c14d4ae389b3a7ca9f5b41519d955a12d7c8dcb9d0bR91

I'd propose to test for the auth dict to be:
* `{"k8s_keychain": True}`
* `{"k8s_keychain": False}`
* `{"secret_name": "thesecret"}`

Reply:
Fixed the default, thanks for that!

I modified the test to pass `auth` as each of these three cases. I'm not sure what you mean by "should be tested when instantiating the class: <link>", as <link> doesn't jump to any specific diff on this PR for me.

Anyway, the entire statement is now covered by the test.

Reply:
Added one more test case for `auth = None`. Now the entire statement is really covered :)

Reply:
I made a fixup commit for the class instantiation test. Think my wording was unclear. Would that work for you?

Reply:
Works for me. I just squashed your fixup commit, thanks!